src/twisted/web/test/test_newclient.py

```diff
@@ -962,6 +961,32 @@ def test_ignored1XXResponseCausesLog(self):
         self.assertEquals(event["log_format"], "Ignoring unexpected {code} response")
         self.assertEquals(event["code"], 103)
 
+    def test_headerShortEnough(self):
+        """Test that HTTP responses with header lines up to 65536 bytes long
+        are allowed."""
+        prefix = b"a: "
+        line = prefix + b"a" * (65536 - len(prefix)) + b"\r\n"
+        response = b"HTTP/1.1 200 OK\r\n" + line + b"\r\n"
+        protocol = HTTPClientParser(
+            Request(b"GET", b"/", _boringHeaders, None), lambda ign: None
+        )
+        protocol.makeConnection(StringTransport())
+        protocol.dataReceived(response)
+        self.assertEqual(protocol.transport.disconnecting, False)
+
+    def test_headerTooLong(self):
+        """Test that HTTP responses with header lines up to 65536 bytes long
+        are allowed."""
```

Comment:
I think that this is a copy/paste error.

Maybe something like this?

```suggestion
        """
        When the remote server returns an HTTP header that has a long line, will disconnected the connection.
        """
```

Is disconnection everything that it does?

I was expecting to also see an error and the reason for disconnection.

Is it possible to also test the error message in this test?
In yes, it wouldhelp to update the docstring.

Reply:
I was actually originally going to ask about that. I was first checking by testing the resulting headers, which I will probably restore in combination with the disconnect check based on https://github.com/twisted/twisted/pull/12094#discussion_r1473704052, and was going to ask for a better way to check that things failed, as I don’t know. The relevant code [simply calls “loseConnection” on the transport](https://github.com/twisted/twisted/blob/234f3788b60f21ccdc51e7b865d244b9343877bf/src/twisted/protocols/basic.py#L541) as far as I can tell.